src/transformers/models/swin/modeling_swin.py

```diff
@@ -522,28 +703,30 @@ def __init__(self, config, dim, input_resolution, depth, num_heads, drop_path, d
 
         self.pointing = False
 
-    def forward(self, hidden_states, head_mask=None, output_attentions=False, output_hidden_states=False):
-        all_hidden_states = () if output_hidden_states else None
-
+    def forward(self, hidden_states, input_dimensions, head_mask=None, output_attentions=False):
+        height, width = input_dimensions
+        block_attentions = () if output_attentions else None
```

Comment:
You can rename `SwinLayer` to `SwinStage` and `SwinBlock` to `SwinLayer` in this PR, such that you can use the term `layer_attentions` here.